### PR TITLE
[DEV-1700] strapi lateral menu 

### PR DESCRIPTION
**Pull request by tommaso1**
#### List of Changes

Adds lateral menu on strapi tutorials

#### Motivation and Context

The content of strapi tutorial is dynamic, therefore we needed a dynamic menu component to extract headings from the page and used them to create a menu

#### How Has This Been Tested?

Go to a strapi tutorial, like `/pago-pa/tutorials/test-tutorial`. 
Check the lateral menu

#### Screenshots (if appropriate):
Screenshot 2024-07-19 at 11 30 33

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/components/templates/TutorialPageTemplate/TutorialPageTemplate.tsx [hunk 1]
```diff
@@ -78,6 +78,21 @@ const TutorialPageTemplate = ({
...
+                position: 'sticky',
+                top: 200,
+                maxWidth: '270px !important',
+              }}
+              id='lateral-menu'
+            >
+              <Box>
+                <PartRendererMenu parts={parts ?? []} />
+              </Box>
+            </Grid>
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/templates/TutorialPageTemplate/TutorialPageTemplate.tsx:95]: I see some differences with static generated tutorial `GuideInPageMenu` like positioning and behavior. I cannot make a suggestion out of this beacuse some changes involve also part not touched in this PR but i leave u a snippet of this file with my changes.
```import EContainer from '@/editorialComponents/EContainer/EContainer';
import ProductBreadcrumbs from '@/components/atoms/ProductBreadcrumbs/ProductBreadcrumbs';
import { productPageToBreadcrumbs } from '@/helpers/breadcrumbs.helpers';
import { FragmentProvider } from '@/components/organisms/FragmentProvider/FragmentProvider';
import { Box, Typography } from '@mui/material';
import RelatedLinks from '@/components/atoms/RelatedLinks/RelatedLinks';
import ProductLayout from '@/components/organisms/ProductLayout/ProductLayout';
import { StrapiTutorials } from '@/lib/strapi/tutorial';
import { Product } from '@/lib/types/product';
import PartRenderer from '@/components/molecules/PartRenderer/PartRenderer';
import { Part } from '@/lib/types/part';
import PartRendererMenu from '@/components/molecules/PartRendererMenu/PartRendererMenu';

// TODO: Remove once the migration to CMS contents will be completed
type TutorialPageTemplateProps = {
  readonly bannerLinks?: StrapiTutorials['data'][0]['attributes']['bannerLinks'];
  readonly parts?: readonly Part[];
  readonly path: string;
  readonly product?: Product;
  readonly relatedLinks?: StrapiTutorials['data'][0]['attributes']['relatedLinks'];
  readonly title: string;
};

const TutorialPageTemplate = ({
  parts,
  path,
  product,
  relatedLinks,
  title,
}: TutorialPageTemplateProps) => {
  return (
    <ProductLayout
      product={product}
      path={path}
      // bannerLinks={bannerLinks} // TODO: Enable this property once the bannerLinks refactor will be completed
    >
      {product && (
        <EContainer
          sx={{
            // 80px is the height of the product header
            marginTop: '80px',
            marginX: 'auto',
            paddingTop: 3,
          }}
        >
          <ProductBreadcrumbs
            breadcrumbs={[
              ...productPageToBreadcrumbs(product, path, [
                {
                  name: title,
                  path: path,
                },
              ]),
            ]}
          />
        </EContainer>
      )}
      <FragmentProvider>
        <Box
          sx={{
            display: 'flex',
            flexDirection: { xs: 'column', lg: 'row' },
            maxWidth: '1156px',
            margin: '0 auto',
            paddingY: '56px',
          }}
        >
          <Box
            sx={{
              flexGrow: { lg: 1 },
              maxWidth: {
                xs: '100%',
                lg: '822px',
              },
            }}
          >
            <Typography
              component='h1'
              sx={{
                fontSize: '38px',
                fontWeight: 700,
                lineHeight: '42px',
                paddingY: 2,
              }}
            >
              {title}
            </Typography>
            {parts && (
              <Box mt={1}>
                {parts.map((part, index) => (
                  <PartRenderer key={index} part={part} />
                ))}
              </Box>
            )}
          </Box>
          <Box
            sx={{
              display: { xs: 'none', lg: 'initial' },
              position: 'relative',
              // 78px is the height of the header, 80px is the height of the product header
              paddingTop: '158px',
              paddingLeft: '64px',
              width: { lg: '270px' },
            }}
          >
            <Box
              sx={{
                position: 'sticky',
                maxWidth: '270px',
                top: 144,
              }}
            >
              <PartRendererMenu parts={parts ?? []} />
            </Box>
          </Box>
        </Box>
      </FragmentProvider>
      {relatedLinks && (
        <RelatedLinks
          title={relatedLinks?.title}
          links={relatedLinks?.links ?? []}
        />
      )}
    </ProductLayout>
  );
};

export default TutorialPageTemplate;

```

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/templates/TutorialPageTemplate/TutorialPageTemplate.tsx (hunk 1 above)
Comment (marcobottaro) [reply apps/nextjs-website/src/components/templates/TutorialPageTemplate/TutorialPageTemplate.tsx]: @tommaso1 I think that @MarcoPonchia with "behaviour" is referring to what happens when you click on a link on the side menu: when navigating to an anchor link, the title associated with the anchor is hidden by the main menu. 
We solved this issue on guides and manuals pages with the PR #899, by adding a box around the h* tags with a positive padding top and a negative margin top of the same size and by moving the id to the surrounding box.
Is there a way to do the same when dealing with the titles generated by the Strapi part renderer?

File: .changeset/six-ants-attack.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"nextjs-website": minor
+---
+
+add lateral menu on strapi tutorials
```
Comment (marcobottaro) [inline .changeset/six-ants-attack.md:5]:
```suggestion
Add lateral menu on Strapi tutorials
```

File: apps/nextjs-website/src/components/molecules/BlocksRendererClient/BlocksRendererClient.tsx [hunk 3]
```diff
@@ -56,7 +57,12 @@ const BlocksRendererClient = ({
...
         heading: ({ children, level }) => (
-          <Typography marginY={4} variant={`h${level}`} color={textColor}>
+          <Typography
+            marginY={4}
+            variant={`h${level}`}
+            color={textColor}
+            id={computeId('blockRenderer', children)}
+          >
             {children}
           </Typography>
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/molecules/BlocksRendererClient/BlocksRendererClient.tsx:67]: with a div or with a Box

```suggestion
          <div
              id={computeId('blockRenderer', children)}
              style={{ marginTop: `-${scrollOffset}px`, paddingTop: `${scrollOffset}px` }}
          >
                    <Typography
                      marginY={4}
                      variant={`h${level}`}
                      color={textColor}
                    >
                      {children}
                    </Typography>
          </div>
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/templates/TutorialTemplate/TutorialTemplate.tsx [hunk 4]
```diff
@@ -55,41 +55,66 @@ const TutorialTemplate = ({
...
+          >
+            <Box
+              sx={{
+                position: 'sticky',
+                maxWidth: '270px',
+                top: 144,
+              }}
+            >
+              <PartRendererMenu parts={parts ?? []} />
+            </Box>
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/templates/TutorialTemplate/TutorialTemplate.tsx:114]: The Box is not well positioned 
image

the title "IN QUESTA PAGINA" shouldn't be visible if there are no parts with titles
image

cfr. https://www.figma.com/design/fQkOTQmYbcLVOdZ6NRjVXf/24-Q1-e-Q2---DevPortal?node-id=25-497&t=K3avOge6ijrnSjxx-4

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by MarcoPonchia — APPROVED** [review]

**Review by MarcoPonchia — APPROVED** [review]

**Pull request merged**